btrdbextras/dq.py

```diff
@@ -0,0 +1,409 @@
+        # There's no event if there's no data
+        if start is None and end is None:
+            return False
+        width = end - start
+        windows, _ = zip(*self.windows(start, end, width, depth))
+        return any(w.max >= 1 for w in windows)
+
+    def __repr__(self):
+        return f"DQDistillate collection={self.collection}, name={self.name}, type={self.type}"
+
+class DQStream(Stream):
+    """
+    Subsets StreamSet object. Contains an original stream along with its
+    distillate Streams
+
+    Parameters
+    ----------
+    stream: btrdb.stream.Stream
+    """
+    def __init__(self, stream):
+        # gives all same attrs/methods as Stream
+        super().__init__(stream._btrdb, stream.uuid)
+        self._distillates = self._get_distillates()
+		
+    def _get_distillates(self):
+        """
+        Finds distillate Streams for each of the underlying source Streams
+
+        Returns
+        -------
+        list[DQDistillate]
+            list of distillate Streams
+        """
+        # NOTE: This involves looking up distillate streams by their source_uuid annotation, so we
+        # need to make sure that all distillers give output streams this annotation
+        distillates = []
+        for stream in self._btrdb.streams_in_collection(annotations={"source_uuid": str(self.uuid)}):
+            try:
+                distillates.append(DQDistillate(stream._btrdb, stream.uuid))
+            # we only want to raise this error if we aren't sure which distiller the stream is
+            # referring to, it's fine if it's a non data quality distiller, that is expected
+            except ValueError as e:
+                if isinstance(e, AmbiguousDistillateError):
+                    raise
+                else:
+                    continue
+        if len(distillates) < 1:
+            warnings.warn(f"Could not find any data quality distillates for stream {str(self.uuid)}")
+        return distillates
+		
+    @property
+    def distillates(self):
+        """
+        Returns list of distillate Streams
+        """
+        return self._distillates
+        
+    def list_distillates(self, notebook=False):
+        """
+        Outputs table showing which distillates the Stream has available
+
+        Parameters
+        ----------
+        notebook: bool
+            Whether or not this function is run from a notebook. Ensures
+            legible formatting
+
+        Returns
+        -------
+        str
+            Table showing which distillates the Stream has available
+        """
+        fmt = "html" if notebook else None
+        table = [["uuid", "collection", "name"] + KNOWN_DISTILLER_TYPES]
+        temp = [str(self.uuid)[:8] + "...", self.collection, self.name]
+        for distiller in KNOWN_DISTILLER_TYPES:
+            try:
+                _ = self[distiller]
+                temp.append(u'\u2713')
+            except KeyError:
+                temp.append("x")
```

Comment:
good call. making that change!

Reply:
@chrisjryan I did try this, but since I'm using the `__getitem__` method, using `get()` to access a specific distiller did not work. I also deleted this line since I removed the table from `list_distillates()` but I do something similar in the contains issue methods.